cadc-access-control/src/main/java/ca/nrc/cadc/ac/client/GMSClient.java

```diff
@@ -457,55 +452,37 @@ public void deleteGroup(String groupName)
         throws GroupNotFoundException, AccessControlException, IOException
     {
         URL groupsURL = getRegistryClient()
-            .getServiceURL(this.serviceID, Standards.GMS_GROUPS_01, AuthMethod.CERT);
+            .getServiceURL(this.serviceID, Standards.GMS_GROUPS_01, getAuthMethod());
         URL deleteGroupURL = new URL(groupsURL.toExternalForm() + "/" + groupName);
         log.debug("deleteGroup request to " + deleteGroupURL.toString());
 
         // reset the state of the cache
         clearCache();
 
-        HttpURLConnection conn =
-                (HttpURLConnection) deleteGroupURL.openConnection();
-        conn.setRequestMethod("DELETE");
-
-        SSLSocketFactory sf = getSSLSocketFactory();
-        if ((sf != null) && ((conn instanceof HttpsURLConnection)))
-        {
-            ((HttpsURLConnection) conn)
-                    .setSSLSocketFactory(sf);
-        }
-
-        final int responseCode;
-
-        try
-        {
-            responseCode = conn.getResponseCode();
-        }
-        catch(Exception e)
-        {
-            throw new AccessControlException(e.getMessage());
-        }
+        HttpDelete delete = new HttpDelete(deleteGroupURL, true);
+        delete.setSSLSocketFactory(getSSLSocketFactory());
+        delete.run();
 
-        if (responseCode != 200)
+        Throwable error = delete.getThrowable();
+        if (error != null)
         {
-            String errMessage = NetUtil.getErrorBody(conn);
```

Comment:
I'm not sure if the text in getErrorBody() and throwable.getMessage() are the same.  If different, then the error reported on failed deletes will change.  Probably no biggie unless there's an integration test expecting a certain response.